platform-sdk/swirlds-merkledb/src/main/java/com/swirlds/merkledb/files/hashmap/HalfDiskHashMap.java

```diff
@@ -160,14 +166,23 @@ public class HalfDiskHashMap implements AutoCloseable, Snapshotable, FileStatist
     /** Fork-join pool for HDHM.endWriting() */
     private static volatile ForkJoinPool flushingPool = null;
 
-    private static ForkJoinPool getFlushingPool() {
+    /**
+     * This method is invoked from a non-static method and uses the provided configuration.
+     * Consequently, the flushing pool will be initialized using the configuration provided
+     * by the first instance of HalfDiskHashMap class that calls the relevant non-static method.
+     * Subsequent calls will reuse the same pool, regardless of any new configurations provided.
+     * </br>
+     * FUTURE WORK: it can be moved to MerkleDb.
```

Comment:
File a ticket to track it, or leave a comment in the ticket about moving MerkleDb to PlatformContext

Reply:
https://github.com/hashgraph/hedera-services/issues/15945